components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+
+import React from "react";
+import { Table } from "antd";
+import { SearchOutlined } from "@ant-design/icons";
+import { Button, Flex, Tooltip } from "antd";
+import { boolean } from "zod";
+import { EditTwoTone, DeleteTwoTone } from "@ant-design/icons";
+
+interface DataTable {
+  [key: string]: any;
+  flagSetting: boolean;
+}
+
+interface DataTableProps {
+  data: DataTable[];
+}
+
+const CustomTable = ({ data }: DataTableProps) => {
```

Comment:
The component name and its interface props are not consistent. You should follow the format `Component` and `ComponentProps`.